CorsixTH/Lua/objects/doors/swing_door_right.lua

```diff
@@ -157,11 +170,12 @@ function SwingDoor:getWalkableTiles()
 end
 
 function SwingDoor:afterLoad(old, new)
-  if old < 181 and self.is_master and not self.slave then
-    local slave_type, x, y = "swing_door_left", self.tile_x, self.tile_y
-    self.slave = self.world:getObject(x - 1, y, slave_type) or self.world:getObject(x, y - 1, slave_type)
-    self.slave:setAsSlave(self)
-    assert(self.slave.master, "Swing doors are not paired at " .. x .. "," .. y)
+  if old < 184 then
+    if (self.is_master and not self.slave) or (not self.is_master and
+        not self.master) then
+    local x, y =  self.tile_x, self.tile_y
+    self:pairDoors(x, y)
```

Comment:
```suggestion
    self:pairDoors(x, y)
    assert((self.master and self.master.slave) or (self.slave and self.slave.master),
        "Swing doors did not pair at " .. x .. "," .. y)
```
Tested on a newly made save from v0.67 and new worlds.

Reply:
That is probably fine for the afterload, though it will still potentially fall over if the save happened during the door creation (incredibly slim). Would suffer the same fate as https://github.com/CorsixTH/CorsixTH/pull/2510/files#r1509644565 in that case.